### DISCUSSION TITLE
Link to an Extension's README in the Extensions View?

### DISCUSSION THREAD

**Discussion by ericdrobinson** [microsoft/vscode-discussions#24] [Extension Development QnA] [answered]
I would like to be able to provide users with the ability to view the extension's README (its main documentation). This includes in [notification](https://code.visualstudio.com/api/references/vscode-api#window.showInformationMessage) buttons about some [encountered issue](https://github.com/microsoft/vscode-discussions/discussions/19#discussioncomment-2892217).

Is there an API that I can use to send users to the Extensions View, select my extension specifically, and jump the user to a specific header within the README?

**Comment by alefragnani** [comment, marked as answer]
Hi @ericdrobinson ,

There is no such API, today.

The best alternative is `workbench.extensions.action.showExtensionsWithIds`, which will open the Extensions viewlet with the search field filled with the extensions you provided as params to the command. But the user will have to _select it_ in order to display its content/readme in the editor.

If this is not enough for your use case, I suggest you to open an issue on VS Code repo.

Hope this helps

**Reply by ericdrobinson** [reply]
@alefragnani Thanks very much for the response and input.

Also, the command you referenced (`workbench.extensions.action.showExtensionsWithIds`) is not presently documented in the [VS Code API Commands Reference](https://code.visualstudio.com/api/references/commands). There are only three there at this time. Is there a better reference for such commands?

**Reply by alefragnani** [reply]
Glad to hear it helps.

When I need to find some command, I mostly use the `Preferences: Open Default Keyboard Shortcuts (JSON)` command. Then, I look at the official documentation (the link you provided) to better understand what/how a particular command works 😄.

**Reply by ericdrobinson** [reply]
Great note! Would be great to see that tip in the [Commands Reference documentation](https://code.visualstudio.com/api/references/commands) to identify some commands that may have slipped through.

Yeah, without having the documentation it makes it a bit tough to understand what arguments are needed/optional and what exactly they do...

**Reply by digitarald** [reply]
Filed https://github.com/microsoft/vscode-docs/issues/5392 for that one. Happy to get contributions, based on your fresh experience.

**Reply by ericdrobinson** [reply]
I think @alefragnani would be the best person to provide the tip suggested in https://github.com/microsoft/vscode-docs/issues/5392. I haven't actually used that approach myself yet, so I'm not familiar with the workflow.

Regardless, it appears that the [Commands Reference documentation](https://code.visualstudio.com/api/references/commands) is incomplete. Would it not make sense to also add an Issue about auditing the built in commands to add missing ones?

I'm guessing that implementers are not terribly familiar with the Commands Reference documentation so I suspect that there are likely more commands missing from the documentation...
